### ISSUE TITLE
False positive `unreachable` after instantiating `_sitebuiltins.Quitter`

### ISSUE THREAD

**Issue by ekanshul** [PyCQA/pylint#11310]
### Bug description

`utils.is_terminating_func` treats any call whose callee infers to a node with the qualified name `_sitebuiltins.Quitter` as terminating. That is right for `exit()` / `quit()` (the builtins are *instances* of `Quitter`), but it also matches the `Quitter` class itself, so *constructing* a `Quitter` is treated as if the program exits there and the next statement is reported as unreachable.

This happens in the stdlib: `Lib/_pyrepl/simple_interact.py` builds a `REPL_COMMANDS` dict with three `_sitebuiltins.Quitter(...)` entries and pylint reports `unreachable` three times on the `def` that follows it.

```python
import _sitebuiltins

REPL_COMMANDS = {
    "exit": _sitebuiltins.Quitter("exit", ""),
    "quit": _sitebuiltins.Quitter("quit", ""),
}


def g():
    pass
```

### Command used

```shell
pylint --disable=all --enable=unreachable a.py
```

### Pylint output

```shell
************* Module a
a.py:9:0: W0101: Unreachable code (unreachable)
a.py:9:0: W0101: Unreachable code (unreachable)
```

### Expected behavior

No message. Instantiating a `Quitter` does not exit; only calling the instance does.

### Pylint version

```shell
pylint 4.1.0-dev0
astroid 4.4.0-dev0
Python 3.13.11
```

### OS / Environment

macOS
